### PR TITLE
[WOE] Implement Charging Hooligan

### PR DESCRIPTION
**Pull request by Susucre**

### CODE REVIEW

**Comment by Susucre** [comment]
[[Charging Hooligan]]

File: Mage.Sets/src/mage/cards/c/ChargingHooligan.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        super(card);
+    }
+
+    @Override
+    public ChargingHooligan copy() {
+        return new ChargingHooligan(this);
+    }
+}
+
+enum ChargingHooliganCondition implements Condition {
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/c/ChargingHooligan.java:65]: you can just use `PermanentsOnTheBattlefieldCondition` with a filter here

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/c/ChargingHooligan.java (hunk 1 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/c/ChargingHooligan.java:65]: Oh right, decided to not do that as I could not set the message in the `PermanentsOnTheBattlefieldCondition`. (and felt weird to have a Filter message be "a Rat is attacking".
Ended up adding a `staticText` parameter to `PermanentsOnTheBattlefieldCondition`, and using that.

Comment (Susucre) [reply Mage.Sets/src/mage/cards/c/ChargingHooligan.java]: It feels weird to have a filter with text "a Rat is attacking", but very well I'll do that this way.

Comment (theelk801) [reply Mage.Sets/src/mage/cards/c/ChargingHooligan.java]: it doesn't matter, the filter text is only ever being used in that one place

**Pull request merged**